src/metabase/api/embed.clj

```diff
@@ -332,8 +332,6 @@
 (defn- check-embedding-enabled-for-object
   "Check that embedding is enabled, that `object` exists, and embedding for `object` is enabled."
   ([entity id]
-   (api/check (pos-int? id)
-              [400 (tru "Dashboard id should be a positive integer.")])
```

Comment:
I think you can just do it in this function here.

```clojure
[400 (case entity
       :model/Card (tru "Card id should be should be a positive integer")
       :model/Dashboard (tru "Dashboard id should be a positive integer")
       (tru "{0} id should be a positive integer" (name entity)))]
```

Reply:
Thought it made less sense to do a if statement here as we already already know what the entity is in the upstream functions, but don't have strong feelings either way.